### PR TITLE
Properly shared `XML.qll` implementation

### PR DESCRIPTION
**Pull request by hvitved**
Commit-by-commit review is suggested, and the second commit is best reviewed with whitespaces ignored.

### CODE REVIEW

**Review by MathiasVP — APPROVED** [review]
C/C++ 👍 (AFAIK, literally have 0 uses of the XML file in C/C++, so I'm not super concerned about these changes)

**Review by tausbn — APPROVED** [review]
Python looks good to me. 👍

**Review by michaelnebel — APPROVED** [review]
C# LGTM!
Thank you very much for ignore whitespace review recommendation - that made it a lot easier!

**Review by geoffw0 — COMMENTED** [review]
What validation has been done on these changes?  (I ask because I'm suspicious we may be short of tests for this stuff)

Also was there much variation between the implementations in different languages?  At a glance I think the answer is "no", but if there is, I'd like to pay closer attention to those areas.

The CPP changes do look good to me - and this is definitely the kind of code that benefits from being shared. :+1:

**Comment by geoffw0** [comment]
> C/C++ 👍 (AFAIK, literally have 0 uses of the XML file in C/C++, so I'm not super concerned about these changes)

Yes, though may well be customers using this as I understand it.

I should add, the CPP DCA run LGTM as well.  The wobbles are likely noise.

**Comment by hvitved** [comment]
> What validation has been done on these changes? (I ask because I'm suspicious we may be short of tests for this stuff)

Unit tests and DCA.

> Also was there much variation between the implementations in different languages? At a glance I think the answer is "no", but if there is, I'd like to pay closer attention to those areas.

No variation at all; the implementations were previously kept in sync using `identical-files.json`.

**Review by asgerf — APPROVED** [review]

**Review by smowton — APPROVED** [review]
Java changes look plausible

File: java/ql/lib/semmle/code/xml/Ant.qll [hunk 1]
```diff
@@ -9,7 +9,7 @@ class AntTarget extends XmlElement {
   AntTarget() { super.getName() = "target" }
 
   /** Gets the name of this Ant target. */
-  override string getName() { result = this.getAttributeValue("name") }
+  string getName() { result = this.getAttributeValue("name") }
```
Comment (Marcono1234) [inline java/ql/lib/semmle/code/xml/Ant.qll:12]: For future reference: It looks like this accidentally fixed a bug with this class. Previously the `override` repurposed the inherited `getName()`. And combined with the characteristics predicate of `AntTarget` being so unspecific, this actually broke the `XmlElement#getName()` of completely unrelated `<target>` occurrences in XML (even in Maven POMs) it seems.

So it is a good thing that `XmlElement` is now `final`, and that this here is not an `override` anymore.

**Pull request merged**